<!DOCTYPE html><html lang="en"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"><meta name="description" content="Single cell CNVs analysis"><meta name="keywords" content="Single cell"><meta name="author" content="Giftbear"><meta name="copyright" content="Giftbear"><title>Single cell CNVs analysis | Giftbear Notes</title><link rel="shortcut icon" href="/gn.ico"><link rel="stylesheet" href="/css/index.css?version=1.9.1"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/font-awesome@latest/css/font-awesome.min.css?version=1.9.1"><meta name="format-detection" content="telephone=no"><meta http-equiv="x-dns-prefetch-control" content="on"><link rel="dns-prefetch" href="https://cdn.jsdelivr.net"><meta http-equiv="Cache-Control" content="no-transform"><meta http-equiv="Cache-Control" content="no-siteapp"><script>var GLOBAL_CONFIG = { 
  root: '/',
  algolia: undefined,
  localSearch: undefined,
  copy: {
    success: 'Copy successfully',
    error: 'Copy error',
    noSupport: 'The browser does not support'
  },
  hexoVersion: '6.3.0'
} </script><meta name="generator" content="Hexo 6.3.0"></head><body><i class="fa fa-arrow-right" id="toggle-sidebar" aria-hidden="true"></i><div id="sidebar" data-display="true"><div class="toggle-sidebar-info text-center"><span data-toggle="Toggle article">Toggle site</span><hr></div><div class="sidebar-toc"><div class="sidebar-toc__title">Catalog</div><div class="sidebar-toc__progress"><span class="progress-notice">You've read</span><span class="progress-num">0</span><span class="progress-percentage">%</span><div class="sidebar-toc__progress-bar"></div></div><div class="sidebar-toc__content"><ol class="toc"><li class="toc-item toc-level-2"><a class="toc-link" href="#infercnv"><span class="toc-number">1.</span> <span class="toc-text">infercnv</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#Installation"><span class="toc-number">1.1.</span> <span class="toc-text">Installation</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#Usage"><span class="toc-number">1.2.</span> <span class="toc-text">Usage</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#CONICSmat"><span class="toc-number">2.</span> <span class="toc-text">CONICSmat</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#Installation-1"><span class="toc-number">2.1.</span> <span class="toc-text">Installation</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#Usage-1"><span class="toc-number">2.2.</span> <span class="toc-text">Usage</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#CaSpER"><span class="toc-number">3.</span> <span class="toc-text">CaSpER</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#sciCNV"><span class="toc-number">4.</span> <span class="toc-text">sciCNV</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#HoneyBADGER"><span class="toc-number">5.</span> <span class="toc-text">HoneyBADGER</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#RNAseqCNV"><span class="toc-number">6.</span> <span class="toc-text">RNAseqCNV</span></a></li></ol></div></div><div class="author-info hide"><div class="author-info__avatar text-center"><img src="/images/bear.png"></div><div class="author-info__name text-center">Giftbear</div><div class="author-info__description text-center"></div><div class="follow-button"><a target="_blank" rel="noopener" href="https://github.com/giftbear">Follow</a></div><hr><div class="author-info-articles"><a class="author-info-articles__archives article-meta" href="/archives"><span class="pull-left">Articles</span><span class="pull-right">44</span></a><a class="author-info-articles__tags article-meta" href="/tags"><span class="pull-left">Tags</span><span class="pull-right">10</span></a><a class="author-info-articles__categories article-meta" href="/categories"><span class="pull-left">Categories</span><span class="pull-right">10</span></a></div></div></div><div id="content-outer"><div class="no-bg" id="top-container"><div id="page-header"><span class="pull-left"> <a id="site-name" href="/">Giftbear Notes</a></span><i class="fa fa-bars toggle-menu pull-right" aria-hidden="true"></i><span class="pull-right menus">   <a class="site-page" href="/">Home</a><a class="site-page" href="/archives">Archives</a><a class="site-page" href="/categories">Categories</a><a class="site-page" href="/tags">Tags</a></span><span class="pull-right"></span></div><div id="post-info"><div id="post-title">Single cell CNVs analysis</div><div id="post-meta"><time class="post-meta__date"><i class="fa fa-calendar" aria-hidden="true"></i> 2023-11-29</time><span class="post-meta__separator">|</span><i class="fa fa-inbox post-meta__icon" aria-hidden="true"></i><a class="post-meta__categories" href="/categories/Omics/">Omics</a></div></div></div><div class="layout" id="content-inner"><article id="post"><div class="article-container" id="post-content"><h2 id="infercnv"><a href="#infercnv" class="headerlink" title="infercnv"></a>infercnv</h2><h3 id="Installation"><a href="#Installation" class="headerlink" title="Installation"></a>Installation</h3><p><code>BiocManager::install(&quot;infercnv&quot;)</code></p>
<h3 id="Usage"><a href="#Usage" class="headerlink" title="Usage"></a>Usage</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br></pre></td><td class="code"><pre><span class="line">library<span class="punctuation">(</span>infercnv<span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># Count matrix</span></span><br><span class="line">countsMatrix <span class="operator">&lt;-</span> GetAssayData<span class="punctuation">(</span>sobj<span class="punctuation">,</span> assay <span class="operator">=</span> <span class="string">&quot;RNA&quot;</span><span class="punctuation">,</span> slot <span class="operator">=</span> <span class="string">&#x27;counts&#x27;</span><span class="punctuation">)</span></span><br><span class="line"><span class="comment"># Cell annotation</span></span><br><span class="line">cellAnno <span class="operator">&lt;-</span> data.frame<span class="punctuation">(</span>Idents<span class="punctuation">(</span>sobj<span class="punctuation">)</span><span class="punctuation">)</span></span><br><span class="line"><span class="comment"># Gene position file</span></span><br><span class="line">genePos <span class="operator">&lt;-</span> <span class="string">&quot;gene_position.txt&quot;</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># Create infercnv object</span></span><br><span class="line">infercnv_obj <span class="operator">=</span> CreateInfercnvObject<span class="punctuation">(</span>raw_counts_matrix <span class="operator">=</span> countsMatrix<span class="punctuation">,</span> </span><br><span class="line">                                    annotations_file <span class="operator">=</span> cellAnno<span class="punctuation">,</span> </span><br><span class="line">                                    gene_order_file <span class="operator">=</span> genePos<span class="punctuation">,</span></span><br><span class="line">                                    delim <span class="operator">=</span> <span class="string">&quot;\t&quot;</span><span class="punctuation">,</span></span><br><span class="line">                                    chr_exclude <span class="operator">=</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="string">&quot;chrX&quot;</span><span class="punctuation">,</span> <span class="string">&quot;chrY&quot;</span><span class="punctuation">,</span> <span class="string">&quot;chrM&quot;</span><span class="punctuation">)</span><span class="punctuation">,</span></span><br><span class="line">                                    ref_group_names <span class="operator">=</span> <span class="literal">NULL</span><span class="punctuation">)</span></span><br><span class="line"><span class="comment"># If there are normal cells: ref_group_names = c(&quot;T cell&quot;)</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># Infer CNVs</span></span><br><span class="line">infercnv_obj <span class="operator">=</span> infercnv<span class="operator">::</span>run<span class="punctuation">(</span>infercnv_obj<span class="punctuation">,</span></span><br><span class="line">                            cutoff <span class="operator">=</span> <span class="number">0.1</span><span class="punctuation">,</span>                            </span><br><span class="line">                            cluster_by_groups <span class="operator">=</span> <span class="literal">TRUE</span><span class="punctuation">,</span> </span><br><span class="line">                            denoise<span class="operator">=</span><span class="literal">TRUE</span><span class="punctuation">,</span></span><br><span class="line">                            HMM<span class="operator">=</span><span class="literal">TRUE</span><span class="punctuation">,</span></span><br><span class="line">                            out_dir <span class="operator">=</span> <span class="string">&quot;cnv&quot;</span><span class="punctuation">,</span> </span><br><span class="line">                            output_format <span class="operator">=</span> <span class="string">&quot;png&quot;</span><span class="punctuation">,</span></span><br><span class="line">                            num_threads <span class="operator">=</span> <span class="number">30</span><span class="punctuation">)</span></span><br></pre></td></tr></table></figure>
<ol>
<li>How to get gene position file?<ul>
<li>Download gtf_to_position_file.py: <a target="_blank" rel="noopener" href="https://github.com/broadinstitute/infercnv/blob/master/scripts/gtf_to_position_file.py">https://github.com/broadinstitute/infercnv/blob/master/scripts/gtf_to_position_file.py</a></li>
<li><code>python3 gtf_to_position_file.py --attribute_name &quot;gene_name&quot; /opt/refdata-gex-GRCh38-2020-A/genes/genes.gtf gene_position.txt</code></li>
</ul>
</li>
</ol>
<p><em>Reference: <a target="_blank" rel="noopener" href="https://github.com/broadinstitute/infercnv">https://github.com/broadinstitute/infercnv</a></em></p>
<h2 id="CONICSmat"><a href="#CONICSmat" class="headerlink" title="CONICSmat"></a>CONICSmat</h2><h3 id="Installation-1"><a href="#Installation-1" class="headerlink" title="Installation"></a>Installation</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">install.packages<span class="punctuation">(</span><span class="string">&quot;beanplot&quot;</span><span class="punctuation">)</span></span><br><span class="line">install.packages<span class="punctuation">(</span><span class="string">&quot;mixtools&quot;</span><span class="punctuation">)</span></span><br><span class="line">install.packages<span class="punctuation">(</span><span class="string">&quot;pheatmap&quot;</span><span class="punctuation">)</span></span><br><span class="line">install.packages<span class="punctuation">(</span><span class="string">&quot;zoo&quot;</span><span class="punctuation">)</span></span><br><span class="line">install.packages<span class="punctuation">(</span><span class="string">&quot;squash&quot;</span><span class="punctuation">)</span></span><br><span class="line">install.packages<span class="punctuation">(</span><span class="string">&quot;Rtsne&quot;</span><span class="punctuation">)</span></span><br><span class="line">BiocManager<span class="operator">::</span>install<span class="punctuation">(</span><span class="string">&quot;scran&quot;</span><span class="punctuation">)</span></span><br><span class="line">BiocManager<span class="operator">::</span>install<span class="punctuation">(</span><span class="string">&quot;biomaRt&quot;</span><span class="punctuation">)</span></span><br><span class="line">devtools<span class="operator">::</span>install_github<span class="punctuation">(</span><span class="string">&quot;diazlab/CONICS/CONICSmat&quot;</span><span class="punctuation">,</span> dep <span class="operator">=</span> <span class="literal">FALSE</span><span class="punctuation">)</span></span><br></pre></td></tr></table></figure>

<h3 id="Usage-1"><a href="#Usage-1" class="headerlink" title="Usage"></a>Usage</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br></pre></td><td class="code"><pre><span class="line">library<span class="punctuation">(</span>CONICSmat<span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># expr: A genes X cells matrix of log2(CPM/10+1) normalized expression counts</span></span><br><span class="line">gene_pos <span class="operator">&lt;-</span> getGenePositions<span class="punctuation">(</span>rownames<span class="punctuation">(</span>expr<span class="punctuation">)</span><span class="punctuation">)</span></span><br><span class="line">expr <span class="operator">&lt;-</span> filterMatrix<span class="punctuation">(</span>expr<span class="punctuation">,</span>gene_pos<span class="punctuation">[</span><span class="punctuation">,</span><span class="string">&quot;hgnc_symbol&quot;</span><span class="punctuation">]</span><span class="punctuation">,</span>minCells<span class="operator">=</span><span class="number">5</span><span class="punctuation">)</span></span><br><span class="line">normFactor <span class="operator">&lt;-</span> calcNormFactors<span class="punctuation">(</span>expr<span class="punctuation">)</span></span><br><span class="line">regions <span class="operator">&lt;-</span> read.table<span class="punctuation">(</span><span class="string">&quot;chromosome_arm_positions_grch38.txt&quot;</span><span class="punctuation">,</span> sep <span class="operator">=</span> <span class="string">&quot;\t&quot;</span><span class="punctuation">,</span> row.names <span class="operator">=</span> <span class="number">1</span><span class="punctuation">,</span> header <span class="operator">=</span> <span class="built_in">T</span><span class="punctuation">)</span></span><br><span class="line">l <span class="operator">&lt;-</span> plotAll<span class="punctuation">(</span>expr<span class="punctuation">,</span>normFactor<span class="punctuation">,</span>regions<span class="punctuation">,</span>gene_pos<span class="punctuation">,</span><span class="string">&quot;sample&quot;</span><span class="punctuation">)</span></span><br><span class="line">hi <span class="operator">&lt;-</span> plotHistogram<span class="punctuation">(</span>l<span class="punctuation">,</span>expr<span class="punctuation">,</span>clusters<span class="operator">=</span><span class="number">2</span><span class="punctuation">,</span>zscoreThreshold<span class="operator">=</span><span class="number">4</span><span class="punctuation">)</span></span><br><span class="line">lrbic <span class="operator">&lt;-</span> read.table<span class="punctuation">(</span><span class="string">&quot;sample_BIC_LR.txt&quot;</span><span class="punctuation">,</span>sep<span class="operator">=</span><span class="string">&quot;\t&quot;</span><span class="punctuation">,</span>header<span class="operator">=</span><span class="built_in">T</span><span class="punctuation">,</span>row.names<span class="operator">=</span><span class="number">1</span><span class="punctuation">,</span>check.names<span class="operator">=</span><span class="built_in">F</span><span class="punctuation">)</span></span><br><span class="line">colnames<span class="punctuation">(</span>lrbic<span class="punctuation">)</span></span><br><span class="line">candRegions <span class="operator">&lt;-</span> rownames<span class="punctuation">(</span>lrbic<span class="punctuation">)</span><span class="punctuation">[</span>which<span class="punctuation">(</span>lrbic<span class="punctuation">[</span><span class="punctuation">,</span><span class="string">&quot;BIC difference&quot;</span><span class="punctuation">]</span><span class="operator">&gt;</span><span class="number">200</span> <span class="operator">&amp;</span> lrbic<span class="punctuation">[</span><span class="punctuation">,</span><span class="string">&quot;LRT adj. p-val&quot;</span><span class="punctuation">]</span><span class="operator">&lt;</span><span class="number">0.01</span><span class="punctuation">)</span><span class="punctuation">]</span></span><br><span class="line">hi <span class="operator">&lt;-</span> plotHistogram<span class="punctuation">(</span>l<span class="punctuation">[</span><span class="punctuation">,</span>candRegions<span class="punctuation">]</span><span class="punctuation">,</span>expr<span class="punctuation">,</span>clusters<span class="operator">=</span><span class="number">4</span><span class="punctuation">,</span>zscoreThreshold<span class="operator">=</span><span class="number">4</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line">normal <span class="operator">&lt;-</span> which<span class="punctuation">(</span>hi<span class="operator">==</span><span class="number">1</span><span class="punctuation">)</span></span><br><span class="line">tumor <span class="operator">&lt;-</span> which<span class="punctuation">(</span>hi<span class="operator">!=</span><span class="number">1</span><span class="punctuation">)</span></span><br><span class="line">plotChromosomeHeatmap<span class="punctuation">(</span>expr<span class="punctuation">,</span> normal <span class="operator">=</span> normal<span class="punctuation">,</span> plotcells <span class="operator">=</span> tumor<span class="punctuation">,</span> gene_pos <span class="operator">=</span> gene_pos<span class="punctuation">,</span> chr <span class="operator">=</span> <span class="built_in">T</span><span class="punctuation">,</span> windowsize <span class="operator">=</span> <span class="number">121</span><span class="punctuation">,</span> expThresh <span class="operator">=</span> <span class="number">0.2</span><span class="punctuation">,</span> thresh <span class="operator">=</span> <span class="number">1</span><span class="punctuation">)</span></span><br></pre></td></tr></table></figure>

<p><em>Reference: <a target="_blank" rel="noopener" href="https://github.com/diazlab/CONICS">https://github.com/diazlab/CONICS</a></em></p>
<h2 id="CaSpER"><a href="#CaSpER" class="headerlink" title="CaSpER"></a>CaSpER</h2><p><em>Reference: <a target="_blank" rel="noopener" href="https://github.com/akdess/CaSpER">https://github.com/akdess/CaSpER</a></em></p>
<h2 id="sciCNV"><a href="#sciCNV" class="headerlink" title="sciCNV"></a>sciCNV</h2><p><em>Reference: <a target="_blank" rel="noopener" href="https://github.com/TiedemannLab/sciCNV">https://github.com/TiedemannLab/sciCNV</a></em></p>
<h2 id="HoneyBADGER"><a href="#HoneyBADGER" class="headerlink" title="HoneyBADGER"></a>HoneyBADGER</h2><p><em>Reference: <a target="_blank" rel="noopener" href="https://github.com/JEFworks-Lab/HoneyBADGER">https://github.com/JEFworks-Lab/HoneyBADGER</a></em></p>
<h2 id="RNAseqCNV"><a href="#RNAseqCNV" class="headerlink" title="RNAseqCNV"></a>RNAseqCNV</h2><p><em>Reference: <a target="_blank" rel="noopener" href="https://github.com/honzee/RNAseqCNV">https://github.com/honzee/RNAseqCNV</a></em></p>
</div></article><div class="post-copyright"><div class="post-copyright__author"><span class="post-copyright-meta">Author: </span><span class="post-copyright-info"><a href="mailto:undefined">Giftbear</a></span></div><div class="post-copyright__type"><span class="post-copyright-meta">Link: </span><span class="post-copyright-info"><a href="https://giftbear.github.io/2023/11/29/Single_cell_CNVs_analysis/">https://giftbear.github.io/2023/11/29/Single_cell_CNVs_analysis/</a></span></div><div class="post-copyright__notice"><span class="post-copyright-meta">Copyright Notice: </span><span class="post-copyright-info">All articles in this blog are licensed under <a target="_blank" rel="noopener" href="https://creativecommons.org/licenses/by-nc-sa/4.0/">CC BY-NC-SA 4.0</a> unless stating additionally.</span></div></div><div class="post-meta__tag-list"><a class="post-meta__tags" href="/tags/Single-cell/">Single cell</a></div><nav id="pagination"><div class="next-post pull-right"><a href="/2023/10/22/IGV_deployment/"><span>IGV deployment</span><i class="fa fa-chevron-right"></i></a></div></nav></div></div><footer><div class="layout" id="footer"><div class="copyright">&copy;2021 - 2023 By Giftbear</div><div class="framework-info"><span>Driven - </span><a target="_blank" rel="noopener" href="http://hexo.io"><span>Hexo</span></a><span class="footer-separator">|</span><span>Theme - </span><a target="_blank" rel="noopener" href="https://github.com/Molunerfinn/hexo-theme-melody"><span>Melody</span></a></div><div class="busuanzi"><script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script><span id="busuanzi_container_page_pv"><i class="fa fa-file"></i><span id="busuanzi_value_page_pv"></span><span></span></span></div></div></footer><i class="fa fa-arrow-up" id="go-up" aria-hidden="true"></i><script src="https://cdn.jsdelivr.net/npm/animejs@latest/lib/anime.min.js"></script><script src="https://cdn.jsdelivr.net/npm/jquery@latest/dist/jquery.min.js"></script><script src="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@latest/dist/jquery.fancybox.min.js"></script><script src="https://cdn.jsdelivr.net/npm/velocity-animate@latest/velocity.min.js"></script><script src="https://cdn.jsdelivr.net/npm/velocity-ui-pack@latest/velocity.ui.min.js"></script><script src="/js/utils.js?version=1.9.1"></script><script src="/js/fancybox.js?version=1.9.1"></script><script src="/js/sidebar.js?version=1.9.1"></script><script src="/js/copy.js?version=1.9.1"></script><script src="/js/fireworks.js?version=1.9.1"></script><script src="/js/transition.js?version=1.9.1"></script><script src="/js/scroll.js?version=1.9.1"></script><script src="/js/head.js?version=1.9.1"></script><script>if(/Android|webOS|iPhone|iPod|iPad|BlackBerry/i.test(navigator.userAgent)) {
  $('#nav').addClass('is-mobile')
  $('footer').addClass('is-mobile')
  $('#top-container').addClass('is-mobile')
}</script></body></html>