<!DOCTYPE html><html lang="en"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"><meta name="description" content="ChIP-Seq Pipeline"><meta name="keywords" content="Epigenome"><meta name="author" content="Giftbear"><meta name="copyright" content="Giftbear"><title>ChIP-Seq Pipeline | Giftbear Notes</title><link rel="shortcut icon" href="/gn.ico"><link rel="stylesheet" href="/css/index.css?version=1.9.1"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/font-awesome@latest/css/font-awesome.min.css?version=1.9.1"><meta name="format-detection" content="telephone=no"><meta http-equiv="x-dns-prefetch-control" content="on"><link rel="dns-prefetch" href="https://cdn.jsdelivr.net"><meta http-equiv="Cache-Control" content="no-transform"><meta http-equiv="Cache-Control" content="no-siteapp"><script>var GLOBAL_CONFIG = { 
  root: '/',
  algolia: undefined,
  localSearch: undefined,
  copy: {
    success: 'Copy successfully',
    error: 'Copy error',
    noSupport: 'The browser does not support'
  },
  hexoVersion: '6.3.0'
} </script><meta name="generator" content="Hexo 6.3.0"></head><body><i class="fa fa-arrow-right" id="toggle-sidebar" aria-hidden="true"></i><div id="sidebar" data-display="true"><div class="toggle-sidebar-info text-center"><span data-toggle="Toggle article">Toggle site</span><hr></div><div class="sidebar-toc"><div class="sidebar-toc__title">Catalog</div><div class="sidebar-toc__progress"><span class="progress-notice">You've read</span><span class="progress-num">0</span><span class="progress-percentage">%</span><div class="sidebar-toc__progress-bar"></div></div><div class="sidebar-toc__content"><ol class="toc"><li class="toc-item toc-level-2"><a class="toc-link" href="#Quality-Control%E2%80%93Fastp"><span class="toc-number">1.</span> <span class="toc-text">Quality Control–Fastp</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Alignment%E2%80%93Bowtie2"><span class="toc-number">2.</span> <span class="toc-text">Alignment–Bowtie2</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Bam-Filter"><span class="toc-number">3.</span> <span class="toc-text">Bam Filter</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Bam-to-Bigwig%E2%80%93Deeptools"><span class="toc-number">4.</span> <span class="toc-text">Bam to Bigwig–Deeptools</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Calling-Peaks%E2%80%93MACS3"><span class="toc-number">5.</span> <span class="toc-text">Calling Peaks–MACS3</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Differential-Peaks"><span class="toc-number">6.</span> <span class="toc-text">Differential Peaks</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#Without-replicates%E2%80%93MACS3-bdgdiff"><span class="toc-number">6.1.</span> <span class="toc-text">Without replicates–MACS3 bdgdiff</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#With-replicates%E2%80%93DiffBind"><span class="toc-number">7.</span> <span class="toc-text">With replicates–DiffBind</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Peak-Annotation"><span class="toc-number">8.</span> <span class="toc-text">Peak Annotation</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#Signal-around-gene%E2%80%93Deeptools"><span class="toc-number">8.1.</span> <span class="toc-text">Signal around gene–Deeptools</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#Signal-around-peak-center%E2%80%93Deeptools"><span class="toc-number">8.2.</span> <span class="toc-text">Signal around peak center–Deeptools</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#Peak-annotation%E2%80%93ChIPseeker"><span class="toc-number">8.3.</span> <span class="toc-text">Peak annotation–ChIPseeker</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Calculate-peak-intensity-and-enrichment"><span class="toc-number">9.</span> <span class="toc-text">Calculate peak intensity and enrichment</span></a></li></ol></div></div><div class="author-info hide"><div class="author-info__avatar text-center"><img src="/images/bear.png"></div><div class="author-info__name text-center">Giftbear</div><div class="author-info__description text-center"></div><div class="follow-button"><a target="_blank" rel="noopener" href="https://github.com/giftbear">Follow</a></div><hr><div class="author-info-articles"><a class="author-info-articles__archives article-meta" href="/archives"><span class="pull-left">Articles</span><span class="pull-right">50</span></a><a class="author-info-articles__tags article-meta" href="/tags"><span class="pull-left">Tags</span><span class="pull-right">10</span></a><a class="author-info-articles__categories article-meta" href="/categories"><span class="pull-left">Categories</span><span class="pull-right">10</span></a></div></div></div><div id="content-outer"><div class="no-bg" id="top-container"><div id="page-header"><span class="pull-left"> <a id="site-name" href="/">Giftbear Notes</a></span><i class="fa fa-bars toggle-menu pull-right" aria-hidden="true"></i><span class="pull-right menus">   <a class="site-page" href="/">Home</a><a class="site-page" href="/archives">Archives</a><a class="site-page" href="/categories">Categories</a><a class="site-page" href="/tags">Tags</a></span><span class="pull-right"></span></div><div id="post-info"><div id="post-title">ChIP-Seq Pipeline</div><div id="post-meta"><time class="post-meta__date"><i class="fa fa-calendar" aria-hidden="true"></i> 2024-02-27</time><span class="post-meta__separator">|</span><i class="fa fa-inbox post-meta__icon" aria-hidden="true"></i><a class="post-meta__categories" href="/categories/Omics/">Omics</a></div></div></div><div class="layout" id="content-inner"><article id="post"><div class="article-container" id="post-content"><p>References: </p>
<ol>
<li><a target="_blank" rel="noopener" href="http://biocluster.ucr.edu/~rkaundal/workshops/R_feb2016/ChIPseq/ChIPseq.html">http://biocluster.ucr.edu/~rkaundal/workshops/R_feb2016/ChIPseq/ChIPseq.html</a></li>
<li><a target="_blank" rel="noopener" href="https://github.com/ENCODE-DCC/chip-seq-pipeline2">https://github.com/ENCODE-DCC/chip-seq-pipeline2</a></li>
</ol>
<h2 id="Quality-Control–Fastp"><a href="#Quality-Control–Fastp" class="headerlink" title="Quality Control–Fastp"></a>Quality Control–Fastp</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">fastp \</span><br><span class="line">-i raw_R1.fq.gz -I raw_R2.fq.gz \</span><br><span class="line">-o filter_R1.fq.gz -O filter_R2.fq.gz \</span><br><span class="line">-h sample.QC.html</span><br></pre></td></tr></table></figure>

<h2 id="Alignment–Bowtie2"><a href="#Alignment–Bowtie2" class="headerlink" title="Alignment–Bowtie2"></a>Alignment–Bowtie2</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># Build index</span></span><br><span class="line">bowtie2-build GRCh38.p14.genome.fa /opt/genome/hg38/bowtie2_index</span><br><span class="line"></span><br><span class="line"><span class="comment"># Alignment</span></span><br><span class="line">bowtie2 \</span><br><span class="line">-p 4 -X 2000 \</span><br><span class="line">-x /opt/genome/hg38/bowtie2_index \</span><br><span class="line">-1 filter_R1.fq.gz -2 filter_R2.fq.gz | samtools <span class="built_in">sort</span> -@ 4 -O bam -o sample.raw.bam</span><br></pre></td></tr></table></figure>
<ul>
<li>Bowtie2 installation: <a target="_blank" rel="noopener" href="https://bowtie-bio.sourceforge.net/bowtie2/index.shtml">https://bowtie-bio.sourceforge.net/bowtie2/index.shtml</a> (unzip bowtie2-2.5.3-linux-x86_64.zip)</li>
</ul>
<h2 id="Bam-Filter"><a href="#Bam-Filter" class="headerlink" title="Bam Filter"></a>Bam Filter</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line">samtools view -F 1804 -f 2 -q 30 -u sample.raw.bam | samtools <span class="built_in">sort</span> -n -o sample.f1.bam</span><br><span class="line"></span><br><span class="line">samtools fixmate -r sample.f1.bam sample.fixmate.bam</span><br><span class="line"></span><br><span class="line">samtools view -F 1804 -f 2 -u sample.fixmate.bam | samtools <span class="built_in">sort</span> -o sample.f2.bam</span><br><span class="line"></span><br><span class="line">java -Xmx4G -jar picard.jar MarkDuplicates \</span><br><span class="line">INPUT=sample.f2.bam \</span><br><span class="line">OUTPUT=sample.mkdup.bam \</span><br><span class="line">METRICS_FILE=sample.metri.txt \</span><br><span class="line">VALIDATION_STRINGENCY=LENIENT ASSUME_SORTED=<span class="literal">true</span> REMOVE_DUPLICATES=<span class="literal">false</span></span><br><span class="line"></span><br><span class="line">samtools view -F 1804 -f 2 -b sample.mkdup.bam &gt; sample.filter.bam</span><br><span class="line">samtools index sample.filter.bam</span><br></pre></td></tr></table></figure>

<h2 id="Bam-to-Bigwig–Deeptools"><a href="#Bam-to-Bigwig–Deeptools" class="headerlink" title="Bam to Bigwig–Deeptools"></a>Bam to Bigwig–Deeptools</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">bamCoverage \</span><br><span class="line">-b sample.filter.bam \</span><br><span class="line">-o sample.bw \</span><br><span class="line">-of bigwig --normalizeUsing BPM --binSize 5 --ignoreDuplicates -p 4</span><br></pre></td></tr></table></figure>
<ul>
<li>Deeptools Installation: <figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">conda create -n deeptools python=3.8</span><br><span class="line"><span class="built_in">source</span> activate deeptools</span><br><span class="line">pip install deeptools</span><br></pre></td></tr></table></figure>
(<a target="_blank" rel="noopener" href="https://deeptools.readthedocs.io/en/develop/content/installation.html">https://deeptools.readthedocs.io/en/develop/content/installation.html</a>)</li>
</ul>
<h2 id="Calling-Peaks–MACS3"><a href="#Calling-Peaks–MACS3" class="headerlink" title="Calling Peaks–MACS3"></a>Calling Peaks–MACS3</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">macs3 predictd -i sample.filter.bam <span class="comment"># extsize = (# predicted fragment length is 211 bps from process log)</span></span><br><span class="line"></span><br><span class="line">macs3 callpeak \</span><br><span class="line">-t sample.filter.bam \</span><br><span class="line">-c input.filter.bam \</span><br><span class="line">-n sample \</span><br><span class="line">--outdir /opt/chip/sample \</span><br><span class="line">-f BAMPE -g hs -B --nomodel --extsize 190</span><br></pre></td></tr></table></figure>
<ul>
<li>MACS3 Installation: <code>pip install macs3</code> (<a target="_blank" rel="noopener" href="https://github.com/macs3-project/MACS">https://github.com/macs3-project/MACS</a>)</li>
</ul>
<h2 id="Differential-Peaks"><a href="#Differential-Peaks" class="headerlink" title="Differential Peaks"></a>Differential Peaks</h2><h3 id="Without-replicates–MACS3-bdgdiff"><a href="#Without-replicates–MACS3-bdgdiff" class="headerlink" title="Without replicates–MACS3 bdgdiff"></a>Without replicates–MACS3 bdgdiff</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">macs3 bdgdiff \</span><br><span class="line">--t1 c1_treat_pileup.bdg \</span><br><span class="line">--c1 c1_control_lambda.bdg \</span><br><span class="line">--t2 c2_treat_pileup.bdg \</span><br><span class="line">--c2 c2_control_lambda.bdg \</span><br><span class="line">--d1 17249333 \ <span class="comment"># c1 fragments after filtering in control (From process log)</span></span><br><span class="line">--d2 20109621 \ <span class="comment"># c2 fragments after filtering in control</span></span><br><span class="line">-g 60 -l 190 \</span><br><span class="line">--o-prefix sample</span><br></pre></td></tr></table></figure>
<ul>
<li>Without replicates: Homer getDifferentialPeaks (<a target="_blank" rel="noopener" href="http://homer.ucsd.edu/homer/ngs/mergePeaks.html">http://homer.ucsd.edu/homer/ngs/mergePeaks.html</a>)</li>
</ul>
<h2 id="With-replicates–DiffBind"><a href="#With-replicates–DiffBind" class="headerlink" title="With replicates–DiffBind"></a>With replicates–DiffBind</h2><ul>
<li>DiffBind Installation: <code>conda install -c bioconda bioconductor-diffbind</code></li>
</ul>
<h2 id="Peak-Annotation"><a href="#Peak-Annotation" class="headerlink" title="Peak Annotation"></a>Peak Annotation</h2><h3 id="Signal-around-gene–Deeptools"><a href="#Signal-around-gene–Deeptools" class="headerlink" title="Signal around gene–Deeptools"></a>Signal around gene–Deeptools</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 1. Compute matrix</span></span><br><span class="line">computeMatrix scale-regions \</span><br><span class="line">--skipZeros --missingDataAsZero -p 4 \</span><br><span class="line">-b 3000 -a 3000 \</span><br><span class="line">--regionBodyLength 5000 \</span><br><span class="line">-S H3K27ac.bw H3K4me3.bw INPUT.bw \</span><br><span class="line">-R gencode.v44.annotation.gtf --metagene \</span><br><span class="line">-o gene.gz \</span><br><span class="line">--outFileSortedRegions gene.bed</span><br><span class="line"></span><br><span class="line"><span class="comment"># 2. Plot heatmap</span></span><br><span class="line">plotHeatmap --matrixFile gene.gz -out gene.png --colorMap Blues Blues Greens Greens Oranges --heatmapWidth 8</span><br><span class="line"></span><br><span class="line"><span class="comment"># Get protein coding bed</span></span><br><span class="line">grep <span class="string">&quot;protein_coding&quot;</span> gencode.v44.annotation.gtf | perl -alne <span class="string">&#x27;&#123;next unless $F[2] eq &quot;gene&quot;; /gene_name \&quot;(.*?)\&quot;;/; print &quot;$F[0]\t$F[3]\t$F[4]\t$1&quot;&#125;&#x27;</span> &gt; protein_coding.bed</span><br></pre></td></tr></table></figure>

<h3 id="Signal-around-peak-center–Deeptools"><a href="#Signal-around-peak-center–Deeptools" class="headerlink" title="Signal around peak center–Deeptools"></a>Signal around peak center–Deeptools</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 1. Merge peaks (summits)</span></span><br><span class="line"><span class="built_in">cat</span> *summits.bed &gt; merge_summits.bed</span><br><span class="line"><span class="built_in">sort</span> -k1,1 -k2,2n merge_summits.bed &gt; sort_merge_summits.bed</span><br><span class="line">bedtools merge -i sort_merge_summits.bed &gt; merged_summits.bed</span><br><span class="line"></span><br><span class="line"><span class="comment"># 1. Merge peaks (summits 250bp)</span></span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> `<span class="built_in">ls</span> *summits.bed`; <span class="keyword">do</span> awk <span class="string">&#x27;BGEIN&#123;OFS=&quot;\t&quot;&#125;&#123;print $1&quot;\t&quot;$2-250&quot;\t&quot;$3+250&#125;&#x27;</span> <span class="variable">$i</span> &gt; 250_<span class="variable">$i</span>; <span class="keyword">done</span></span><br><span class="line"><span class="built_in">cat</span> 250*summits.bed | awk <span class="string">&#x27;&#123;if($2&lt;0)&#123;print $1&quot;\t&quot;0&quot;\t&quot;$3&#125; else &#123;print $0&#125;&#125;&#x27;</span> | sortBed -i - | bedtools merge -i - &gt; merged_summits.bed</span><br><span class="line"></span><br><span class="line"><span class="comment"># 2. Compute matrix</span></span><br><span class="line">computeMatrix reference-point --referencePoint center \</span><br><span class="line">--skipZeros --missingDataAsZero -p 4 \</span><br><span class="line">-b 3000 -a 3000 \</span><br><span class="line">-S H3K27ac.bw H3K4me3.bw INPUT.bw \</span><br><span class="line">-R merged_summits.bed \</span><br><span class="line">-o merged_summits_center.gz \</span><br><span class="line">--outFileSortedRegions merged_summits_center.bed</span><br><span class="line"></span><br><span class="line"><span class="comment"># 3. Plot heatmap</span></span><br><span class="line">plotHeatmap --matrixFile merged_summits_center.gz -out merged_summits_center.png --startLabel <span class="string">&quot;Peak Start&quot;</span> --endLabel <span class="string">&quot;Peak End&quot;</span> --xAxisLabel <span class="string">&quot;&quot;</span> --regionsLabel <span class="string">&quot;Peaks&quot;</span> --heatmapWidth 6</span><br></pre></td></tr></table></figure>

<h3 id="Peak-annotation–ChIPseeker"><a href="#Peak-annotation–ChIPseeker" class="headerlink" title="Peak annotation–ChIPseeker"></a>Peak annotation–ChIPseeker</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br></pre></td><td class="code"><pre><span class="line">library<span class="punctuation">(</span>ChIPseeker<span class="punctuation">)</span></span><br><span class="line">library<span class="punctuation">(</span>GenomicFeatures<span class="punctuation">)</span></span><br><span class="line">library<span class="punctuation">(</span>tidyverse<span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line">txdb <span class="operator">&lt;-</span> makeTxDbFromGFF<span class="punctuation">(</span><span class="string">&#x27;gencode.v44.annotation.gff3&#x27;</span><span class="punctuation">)</span></span><br><span class="line">saveDb<span class="punctuation">(</span>txdb<span class="punctuation">,</span> file<span class="operator">=</span><span class="string">&quot;gencode.v44.sqlite&quot;</span><span class="punctuation">)</span></span><br><span class="line">hg38 <span class="operator">&lt;-</span> loadDb<span class="punctuation">(</span><span class="string">&quot;gencode.v44.sqlite&quot;</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line">samplefiles <span class="operator">&lt;-</span> list.files<span class="punctuation">(</span><span class="string">&quot;.&quot;</span><span class="punctuation">,</span> pattern <span class="operator">=</span> <span class="string">&quot;.broadPeak&quot;</span><span class="punctuation">,</span> full.names <span class="operator">=</span> <span class="built_in">T</span><span class="punctuation">)</span></span><br><span class="line">samplefiles <span class="operator">&lt;-</span> as.list<span class="punctuation">(</span>samplefiles<span class="punctuation">)</span></span><br><span class="line"><span class="built_in">names</span><span class="punctuation">(</span>samplefiles<span class="punctuation">)</span> <span class="operator">&lt;-</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="string">&quot;H3K27ac&quot;</span><span class="punctuation">,</span> <span class="string">&quot;H3K4me3&quot;</span><span class="punctuation">)</span></span><br><span class="line">peakAnnoList <span class="operator">&lt;-</span> lapply<span class="punctuation">(</span>samplefiles<span class="punctuation">,</span> annotatePeak<span class="punctuation">,</span> TxDb<span class="operator">=</span>hg38<span class="punctuation">,</span> </span><br><span class="line">                       tssRegion<span class="operator">=</span><span class="built_in">c</span><span class="punctuation">(</span><span class="operator">-</span><span class="number">3000</span><span class="punctuation">,</span> <span class="number">3000</span><span class="punctuation">)</span><span class="punctuation">,</span> addFlankGeneInfo <span class="operator">=</span> <span class="literal">TRUE</span><span class="punctuation">,</span> flankDistance <span class="operator">=</span> <span class="number">5000</span><span class="punctuation">)</span></span><br><span class="line">H3K27ac_anno <span class="operator">&lt;-</span> as.data.frame<span class="punctuation">(</span>peakAnnoList<span class="punctuation">[[</span><span class="string">&quot;H3K27ac&quot;</span><span class="punctuation">]</span><span class="punctuation">]</span><span class="operator">@</span>anno<span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line">tx2gene <span class="operator">&lt;-</span> read.table<span class="punctuation">(</span><span class="string">&quot;gene2gene.txt&quot;</span><span class="punctuation">,</span> header <span class="operator">=</span> <span class="built_in">T</span><span class="punctuation">)</span></span><br><span class="line">tx2gene <span class="operator">&lt;-</span> tx2gene <span class="operator">%&gt;%</span> as.data.frame<span class="punctuation">(</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line">H3K27ac_gi <span class="operator">&lt;-</span> unique<span class="punctuation">(</span>H3K27ac_anno<span class="operator">$</span>geneId<span class="punctuation">)</span></span><br><span class="line">H3K27ac_gs <span class="operator">&lt;-</span> tx2gene <span class="operator">%&gt;%</span> </span><br><span class="line">  filter<span class="punctuation">(</span>geneId <span class="operator">%in%</span> H3K27ac_gi<span class="punctuation">)</span></span><br><span class="line">m <span class="operator">&lt;-</span> match<span class="punctuation">(</span>H3K27ac_anno<span class="operator">$</span>geneId<span class="punctuation">,</span> H3K27ac_gs<span class="operator">$</span>geneId<span class="punctuation">)</span></span><br><span class="line">H3K27ac_anno <span class="operator">&lt;-</span> cbind<span class="punctuation">(</span>H3K27ac_anno<span class="punctuation">[</span><span class="punctuation">,</span><span class="number">1</span><span class="operator">:</span><span class="number">18</span><span class="punctuation">]</span><span class="punctuation">,</span> geneSymbol<span class="operator">=</span>H3K27ac_gs<span class="operator">$</span>symbol<span class="punctuation">[</span>m<span class="punctuation">]</span><span class="punctuation">,</span> H3K27ac_anno<span class="punctuation">[</span><span class="punctuation">,</span><span class="number">19</span><span class="operator">:</span><span class="number">21</span><span class="punctuation">]</span><span class="punctuation">)</span></span><br><span class="line">write.table<span class="punctuation">(</span>H3K27ac_anno<span class="punctuation">,</span> file<span class="operator">=</span><span class="string">&quot;H3K27ac_peakAnn.txt&quot;</span><span class="punctuation">,</span> sep<span class="operator">=</span><span class="string">&quot;\t&quot;</span><span class="punctuation">,</span> <span class="built_in">quote</span><span class="operator">=</span><span class="built_in">F</span><span class="punctuation">,</span> row.names<span class="operator">=</span><span class="built_in">F</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line">peak1 <span class="operator">&lt;-</span> readPeakFile<span class="punctuation">(</span><span class="string">&#x27;sample1_peaks.narrowPeak&#x27;</span><span class="punctuation">)</span></span><br><span class="line">peak2 <span class="operator">&lt;-</span> readPeakFile<span class="punctuation">(</span><span class="string">&#x27;sample2_peaks.narrowPeak&#x27;</span><span class="punctuation">)</span></span><br><span class="line">peaks <span class="operator">&lt;-</span> <span class="built_in">list</span><span class="punctuation">(</span>sample1 <span class="operator">=</span> peak1<span class="punctuation">,</span> sample2 <span class="operator">=</span> peak2<span class="punctuation">)</span></span><br><span class="line">plotPeakProf2<span class="punctuation">(</span>peaks<span class="punctuation">,</span> upstream <span class="operator">=</span> <span class="number">3000</span><span class="punctuation">,</span> downstream <span class="operator">=</span> <span class="number">3000</span><span class="punctuation">,</span> conf <span class="operator">=</span> <span class="number">0.95</span><span class="punctuation">,</span></span><br><span class="line">              by <span class="operator">=</span> <span class="string">&quot;gene&quot;</span><span class="punctuation">,</span> type <span class="operator">=</span> <span class="string">&quot;start_site&quot;</span><span class="punctuation">,</span> TxDb <span class="operator">=</span> hg38<span class="punctuation">,</span> facet<span class="operator">=</span><span class="string">&quot;row&quot;</span><span class="punctuation">)</span></span><br><span class="line">plotPeakProf2<span class="punctuation">(</span>peaks<span class="punctuation">,</span> upstream <span class="operator">=</span> rel<span class="punctuation">(</span><span class="number">0.2</span><span class="punctuation">)</span><span class="punctuation">,</span> downstream <span class="operator">=</span> rel<span class="punctuation">(</span><span class="number">0.2</span><span class="punctuation">)</span><span class="punctuation">,</span></span><br><span class="line">              conf <span class="operator">=</span> <span class="number">0.95</span><span class="punctuation">,</span> by <span class="operator">=</span> <span class="string">&quot;gene&quot;</span><span class="punctuation">,</span> type <span class="operator">=</span> <span class="string">&quot;body&quot;</span><span class="punctuation">,</span> nbin <span class="operator">=</span> <span class="number">800</span><span class="punctuation">,</span></span><br><span class="line">              TxDb <span class="operator">=</span> hg38<span class="punctuation">,</span> facet<span class="operator">=</span><span class="string">&quot;row&quot;</span><span class="punctuation">)</span></span><br><span class="line">peakAnnoList <span class="operator">&lt;-</span> lapply<span class="punctuation">(</span>peaks<span class="punctuation">,</span> annotatePeak<span class="punctuation">,</span> TxDb<span class="operator">=</span>hg38<span class="punctuation">,</span></span><br><span class="line">                       tssRegion<span class="operator">=</span><span class="built_in">c</span><span class="punctuation">(</span><span class="operator">-</span><span class="number">3000</span><span class="punctuation">,</span> <span class="number">3000</span><span class="punctuation">)</span><span class="punctuation">,</span> annoDb<span class="operator">=</span><span class="string">&quot;org.Hs.eg.db&quot;</span><span class="punctuation">)</span> <span class="comment"># annoDb=&quot;org.Hs.eg.db&quot; will produce the gene symbol</span></span><br><span class="line">plotAnnoBar<span class="punctuation">(</span>peakAnnoList<span class="punctuation">)</span></span><br><span class="line">plotDistToTSS<span class="punctuation">(</span>peakAnnoList<span class="punctuation">)</span></span><br><span class="line">upsetplot<span class="punctuation">(</span>peakAnnoList<span class="punctuation">[</span><span class="number">1</span><span class="punctuation">]</span><span class="punctuation">,</span> vennpie<span class="operator">=</span><span class="literal">TRUE</span><span class="punctuation">)</span></span><br><span class="line">write.table<span class="punctuation">(</span>peakAnnoList<span class="punctuation">[</span><span class="number">1</span><span class="punctuation">]</span><span class="punctuation">,</span> file <span class="operator">=</span> <span class="string">&#x27;sample1_peak_ann.txt&#x27;</span><span class="punctuation">,</span> sep <span class="operator">=</span> <span class="string">&#x27;\t&#x27;</span><span class="punctuation">,</span> <span class="built_in">quote</span> <span class="operator">=</span> <span class="literal">FALSE</span><span class="punctuation">,</span> row.names <span class="operator">=</span> <span class="literal">FALSE</span><span class="punctuation">)</span></span><br></pre></td></tr></table></figure>
<ul>
<li>ChIPseeker Installation: <code>BiocManager::install(&quot;ChIPseeker&quot;)</code></li>
<li>Get gene2gene.txt: <code>grep &quot;transcript_id&quot; gencode.v44.annotation.gtf |cut -f 9|cut -f 1,2,4 -d &quot;;&quot; |cut -f 2,6 -d &quot; &quot;|sed &#39;s/&quot;//g&#39; |sed &#39;s/; /\t/g&#39; |uniq &gt; gene2gene.txt</code></li>
<li>Reference: <a target="_blank" rel="noopener" href="https://hbctraining.github.io/In-depth-NGS-Data-Analysis-Course/sessionV/lessons/12_annotation_functional_analysis.html">https://hbctraining.github.io/In-depth-NGS-Data-Analysis-Course/sessionV/lessons/12_annotation_functional_analysis.html</a></li>
</ul>
<h2 id="Calculate-peak-intensity-and-enrichment"><a href="#Calculate-peak-intensity-and-enrichment" class="headerlink" title="Calculate peak intensity and enrichment"></a>Calculate peak intensity and enrichment</h2><ol>
<li><code>gunzip center.mat.gz</code></li>
<li><code>sed -i &#39;1d&#39; center.mat</code></li>
<li><code>cut -f 297-317 center.mat|awk &#39;&#123;sum = 0; for (i = 1; i &lt;= NF; i++) sum += $i; sum /= NF; print sum&#125;&#39; &gt; sample1_intensity.txt</code></li>
<li><code>cut -f 197-297,317-417 center.mat|awk &#39;&#123;sum = 0; for (i = 1; i &lt;= NF; i++) sum += $i; sum /= NF; print sum&#125;&#39; &gt; sample1_enrichment.txt</code></li>
<li><code>paste sample1_intensity.txt sample1_enrichment.txt &gt; temp1.txt</code></li>
<li><code>awk &#39;&#123;if($2==0)&#123;print $1&quot;\t0&quot;&#125;else&#123;print $1&quot;\t&quot;$1/$2&#125;&#125;&#39; temp1.txt &gt; sample1.txt</code></li>
<li><code>cut -f 1-4 center.mat &gt; rowname.txt</code></li>
<li><code>paste rowname.txt sample1.txt &gt; peak_intensity.txt</code></li>
</ol>
</div></article><div class="post-copyright"><div class="post-copyright__author"><span class="post-copyright-meta">Author: </span><span class="post-copyright-info"><a href="mailto:undefined">Giftbear</a></span></div><div class="post-copyright__type"><span class="post-copyright-meta">Link: </span><span class="post-copyright-info"><a href="https://giftbear.github.io/2024/02/27/ChIP-Seq_Pipeline/">https://giftbear.github.io/2024/02/27/ChIP-Seq_Pipeline/</a></span></div><div class="post-copyright__notice"><span class="post-copyright-meta">Copyright Notice: </span><span class="post-copyright-info">All articles in this blog are licensed under <a target="_blank" rel="noopener" href="https://creativecommons.org/licenses/by-nc-sa/4.0/">CC BY-NC-SA 4.0</a> unless stating additionally.</span></div></div><div class="post-meta__tag-list"><a class="post-meta__tags" href="/tags/Epigenome/">Epigenome</a></div><nav id="pagination"><div class="prev-post pull-left"><a href="/2024/03/15/CUTTag_Pipeline/"><i class="fa fa-chevron-left">  </i><span>CUTTag Pipeline</span></a></div><div class="next-post pull-right"><a href="/2023/12/01/DESeq2%E5%B7%AE%E5%BC%82%E8%A1%A8%E8%BE%BE%E5%88%86%E6%9E%90/"><span>DESeq2差异表达分析</span><i class="fa fa-chevron-right"></i></a></div></nav></div></div><footer><div class="layout" id="footer"><div class="copyright">&copy;2021 - 2024 By Giftbear</div><div class="framework-info"><span>Driven - </span><a target="_blank" rel="noopener" href="http://hexo.io"><span>Hexo</span></a><span class="footer-separator">|</span><span>Theme - </span><a target="_blank" rel="noopener" href="https://github.com/Molunerfinn/hexo-theme-melody"><span>Melody</span></a></div><div class="busuanzi"><script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script><span id="busuanzi_container_page_pv"><i class="fa fa-file"></i><span id="busuanzi_value_page_pv"></span><span></span></span></div></div></footer><i class="fa fa-arrow-up" id="go-up" aria-hidden="true"></i><script src="https://cdn.jsdelivr.net/npm/animejs@latest/lib/anime.min.js"></script><script src="https://cdn.jsdelivr.net/npm/jquery@latest/dist/jquery.min.js"></script><script src="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@latest/dist/jquery.fancybox.min.js"></script><script src="https://cdn.jsdelivr.net/npm/velocity-animate@latest/velocity.min.js"></script><script src="https://cdn.jsdelivr.net/npm/velocity-ui-pack@latest/velocity.ui.min.js"></script><script src="/js/utils.js?version=1.9.1"></script><script src="/js/fancybox.js?version=1.9.1"></script><script src="/js/sidebar.js?version=1.9.1"></script><script src="/js/copy.js?version=1.9.1"></script><script src="/js/fireworks.js?version=1.9.1"></script><script src="/js/transition.js?version=1.9.1"></script><script src="/js/scroll.js?version=1.9.1"></script><script src="/js/head.js?version=1.9.1"></script><script>if(/Android|webOS|iPhone|iPod|iPad|BlackBerry/i.test(navigator.userAgent)) {
  $('#nav').addClass('is-mobile')
  $('footer').addClass('is-mobile')
  $('#top-container').addClass('is-mobile')
}</script></body></html>