<!DOCTYPE html><html lang="en"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"><meta name="description" content="CUTTag Pipeline"><meta name="keywords" content="Epigenome"><meta name="author" content="Giftbear"><meta name="copyright" content="Giftbear"><title>CUTTag Pipeline | Giftbear Notes</title><link rel="shortcut icon" href="/gn.ico"><link rel="stylesheet" href="/css/index.css?version=1.9.1"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/font-awesome@latest/css/font-awesome.min.css?version=1.9.1"><meta name="format-detection" content="telephone=no"><meta http-equiv="x-dns-prefetch-control" content="on"><link rel="dns-prefetch" href="https://cdn.jsdelivr.net"><meta http-equiv="Cache-Control" content="no-transform"><meta http-equiv="Cache-Control" content="no-siteapp"><script>var GLOBAL_CONFIG = { 
  root: '/',
  algolia: undefined,
  localSearch: undefined,
  copy: {
    success: 'Copy successfully',
    error: 'Copy error',
    noSupport: 'The browser does not support'
  },
  hexoVersion: '6.3.0'
} </script><meta name="generator" content="Hexo 6.3.0"></head><body><i class="fa fa-arrow-right" id="toggle-sidebar" aria-hidden="true"></i><div id="sidebar" data-display="true"><div class="toggle-sidebar-info text-center"><span data-toggle="Toggle article">Toggle site</span><hr></div><div class="sidebar-toc"><div class="sidebar-toc__title">Catalog</div><div class="sidebar-toc__progress"><span class="progress-notice">You've read</span><span class="progress-num">0</span><span class="progress-percentage">%</span><div class="sidebar-toc__progress-bar"></div></div><div class="sidebar-toc__content"><ol class="toc"><li class="toc-item toc-level-2"><a class="toc-link" href="#Alignment%E2%80%93Bowtie2"><span class="toc-number">1.</span> <span class="toc-text">Alignment–Bowtie2</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Bam-convert-to-bedgraph"><span class="toc-number">2.</span> <span class="toc-text">Bam convert to bedgraph</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Calling-peaks%E2%80%93SEACR"><span class="toc-number">3.</span> <span class="toc-text">Calling peaks–SEACR</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Visualization%E2%80%93Deeptools"><span class="toc-number">4.</span> <span class="toc-text">Visualization–Deeptools</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#Bam-convert-to-bigwig"><span class="toc-number">4.1.</span> <span class="toc-text">Bam convert to bigwig</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#Signal-around-gene"><span class="toc-number">4.2.</span> <span class="toc-text">Signal around gene</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#Signal-around-peak-center"><span class="toc-number">4.3.</span> <span class="toc-text">Signal around peak center</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Peak-annotation%E2%80%93ChIPseeker"><span class="toc-number">5.</span> <span class="toc-text">Peak annotation–ChIPseeker</span></a></li></ol></div></div><div class="author-info hide"><div class="author-info__avatar text-center"><img src="/images/bear.png"></div><div class="author-info__name text-center">Giftbear</div><div class="author-info__description text-center"></div><div class="follow-button"><a target="_blank" rel="noopener" href="https://github.com/giftbear">Follow</a></div><hr><div class="author-info-articles"><a class="author-info-articles__archives article-meta" href="/archives"><span class="pull-left">Articles</span><span class="pull-right">53</span></a><a class="author-info-articles__tags article-meta" href="/tags"><span class="pull-left">Tags</span><span class="pull-right">10</span></a><a class="author-info-articles__categories article-meta" href="/categories"><span class="pull-left">Categories</span><span class="pull-right">10</span></a></div></div></div><div id="content-outer"><div class="no-bg" id="top-container"><div id="page-header"><span class="pull-left"> <a id="site-name" href="/">Giftbear Notes</a></span><i class="fa fa-bars toggle-menu pull-right" aria-hidden="true"></i><span class="pull-right menus">   <a class="site-page" href="/">Home</a><a class="site-page" href="/archives">Archives</a><a class="site-page" href="/categories">Categories</a><a class="site-page" href="/tags">Tags</a></span><span class="pull-right"></span></div><div id="post-info"><div id="post-title">CUTTag Pipeline</div><div id="post-meta"><time class="post-meta__date"><i class="fa fa-calendar" aria-hidden="true"></i> 2024-03-15</time><span class="post-meta__separator">|</span><i class="fa fa-inbox post-meta__icon" aria-hidden="true"></i><a class="post-meta__categories" href="/categories/Omics/">Omics</a></div></div></div><div class="layout" id="content-inner"><article id="post"><div class="article-container" id="post-content"><p>References: <a target="_blank" rel="noopener" href="https://yezhengstat.github.io/CUTTag_tutorial/">https://yezhengstat.github.io/CUTTag_tutorial/</a></p>
<h2 id="Alignment–Bowtie2"><a href="#Alignment–Bowtie2" class="headerlink" title="Alignment–Bowtie2"></a>Alignment–Bowtie2</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># Build hg38 index</span></span><br><span class="line">bowtie2-build GRCh38.p14.genome.fa /opt/database/hg38</span><br><span class="line"><span class="comment"># Build E.coli index</span></span><br><span class="line">bowtie2-build Escherichia_coli_str_k_12_substr_mg1655_gca_000005845.ASM584v2.dna.chromosome.Chromosome.fa /opt/database/ecoli</span><br><span class="line"></span><br><span class="line"><span class="comment"># Alignment</span></span><br><span class="line">bowtie2 --end-to-end --very-sensitive --no-mixed --no-discordant --phred33 \</span><br><span class="line">-I 10 -X 700 -p 4 \</span><br><span class="line">-x /opt/database/hg38 \</span><br><span class="line">-1 sample.R1.fastq.gz -2 sample.R2.fastq.gz \</span><br><span class="line">-S sample.sam &amp;&gt; sample.txt</span><br><span class="line"></span><br><span class="line">bowtie2 --end-to-end --very-sensitive --no-mixed --no-discordant --phred33 \</span><br><span class="line">-I 10 -X 700 -p 4 \</span><br><span class="line">-x /opt/database/ecoli \</span><br><span class="line">-1 sample.R1.fastq.gz -2 sample.R2.fastq.gz \</span><br><span class="line">-S sample.spikeIn.sam &amp;&gt; sample.spikeIn.txt</span><br></pre></td></tr></table></figure>
<ul>
<li>E.coli genome download: <a target="_blank" rel="noopener" href="https://ftp.ensemblgenomes.ebi.ac.uk/pub/bacteria/release-58/fasta/bacteria_0_collection/escherichia_coli_str_k_12_substr_mg1655_gca_000005845/dna/">https://ftp.ensemblgenomes.ebi.ac.uk/pub/bacteria/release-58/fasta/bacteria_0_collection/escherichia_coli_str_k_12_substr_mg1655_gca_000005845/dna/</a></li>
</ul>
<h2 id="Bam-convert-to-bedgraph"><a href="#Bam-convert-to-bedgraph" class="headerlink" title="Bam convert to bedgraph"></a>Bam convert to bedgraph</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><span class="line">samtools view -bS -F 0x04 sample.sam &gt; sample.mapped.bam</span><br><span class="line">bedtools bamtobed -bedpe -i sample.mapped.bam &gt; sample.bed</span><br><span class="line">awk <span class="string">&#x27;$1==$4 &amp;&amp; $6-$2 &lt; 1000 &#123;print $0&#125;&#x27;</span> sample.bed &gt; sample.clean.bed</span><br><span class="line"><span class="built_in">cut</span> -f 1,2,6 sample.clean.bed | <span class="built_in">sort</span> -k1,1 -k2,2n -k3,3n &gt; sample.fragments.bed</span><br><span class="line"></span><br><span class="line"><span class="comment"># Spike-in calibration (Optional)</span></span><br><span class="line">seqDepthDouble=`samtools view -F 0x04 sample.spikeIn.sam | <span class="built_in">wc</span> -l`</span><br><span class="line">seqDepth=$((seqDepthDouble/<span class="number">2</span>))</span><br><span class="line"><span class="built_in">echo</span> <span class="variable">$seqDepth</span> &gt; sample.spikeIn.seqDepth</span><br><span class="line"></span><br><span class="line">seqDepth=`<span class="built_in">cat</span> sample.spikeIn.seqDepth`</span><br><span class="line"><span class="keyword">if</span> [[ <span class="string">&quot;<span class="variable">$seqDepth</span>&quot;</span> -gt <span class="string">&quot;1&quot;</span> ]]</span><br><span class="line"><span class="keyword">then</span></span><br><span class="line">    scale_factor=`<span class="built_in">echo</span> <span class="string">&quot;10000 / <span class="variable">$seqDepth</span>&quot;</span> | bc -l`</span><br><span class="line">    bedtools genomecov -<span class="built_in">bg</span> -scale <span class="variable">$scale_factor</span> \</span><br><span class="line">    -i sample.fragments.bed \</span><br><span class="line">    -g GRCh38.p14.genome.size &gt; sample.fragments.normalized.bedgraph</span><br><span class="line"><span class="keyword">fi</span></span><br></pre></td></tr></table></figure>
<ul>
<li>Get GRCh38.p14.genome.size: <code>samtools faidx GRCh38.p14.genome.fa</code>;<code>cut -f 1,2 GRCh38.p14.genome.fa.fai &gt; GRCh38.p14.genome.size</code></li>
</ul>
<h2 id="Calling-peaks–SEACR"><a href="#Calling-peaks–SEACR" class="headerlink" title="Calling peaks–SEACR"></a>Calling peaks–SEACR</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">bash /opt/SEACR-master/SEACR_1.3.sh \</span><br><span class="line">sample.fragments.normalized.bedgraph \</span><br><span class="line">0.01 \</span><br><span class="line">non stringent sample_seacr_peak</span><br></pre></td></tr></table></figure>
<ul>
<li>SEACR download: <a target="_blank" rel="noopener" href="https://github.com/FredHutch/SEACR/">https://github.com/FredHutch/SEACR/</a></li>
</ul>
<h2 id="Visualization–Deeptools"><a href="#Visualization–Deeptools" class="headerlink" title="Visualization–Deeptools"></a>Visualization–Deeptools</h2><h3 id="Bam-convert-to-bigwig"><a href="#Bam-convert-to-bigwig" class="headerlink" title="Bam convert to bigwig"></a>Bam convert to bigwig</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">samtools <span class="built_in">sort</span> -o sample.sorted.bam sample.mapped.bam                                                     </span><br><span class="line">samtools index sample.sorted.bam</span><br><span class="line">bamCoverage -b sample.sorted.bam -o sample.bw -of bigwig --normalizeUsing BPM --binSize 5 --ignoreDuplicates -p 8</span><br></pre></td></tr></table></figure>

<h3 id="Signal-around-gene"><a href="#Signal-around-gene" class="headerlink" title="Signal around gene"></a>Signal around gene</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">computeMatrix scale-regions -S sample.bw \</span><br><span class="line">                            -R /opt/database/gencode.v44.annotation.gtf \</span><br><span class="line">                            --beforeRegionStartLength 3000 \</span><br><span class="line">                            --regionBodyLength 5000 \</span><br><span class="line">                            --afterRegionStartLength 3000 \</span><br><span class="line">                            --skipZeros --missingDataAsZero --metagene -p 8 \</span><br><span class="line">                            -o gene_mat.gz</span><br><span class="line"></span><br><span class="line">plotHeatmap -m gene_mat.gz -out gene_enrichment.png --sortUsing <span class="built_in">sum</span> --heatmapWidth 8</span><br></pre></td></tr></table></figure>

<h3 id="Signal-around-peak-center"><a href="#Signal-around-peak-center" class="headerlink" title="Signal around peak center"></a>Signal around peak center</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">awk <span class="string">&#x27;&#123;split($6, summit, &quot;:&quot;); split(summit[2], region, &quot;-&quot;); print summit[1]&quot;\t&quot;region[1]&quot;\t&quot;region[2]&#125;&#x27;</span> sample_seacr_peak.stringent.bed &gt; sample.summitRegion.bed</span><br><span class="line"></span><br><span class="line">computeMatrix reference-point --referencePoint center \</span><br><span class="line">                              -S sample.bw \</span><br><span class="line">                              -R sample.summitRegion.bed \</span><br><span class="line">                              -a 3000 -b 3000 \</span><br><span class="line">                              --skipZeros --missingDataAsZero -p 8 \</span><br><span class="line">                              -o sample_mat.gz </span><br><span class="line">plotHeatmap -m sample_mat.gz -out sample_heatmap.png --sortUsing <span class="built_in">sum</span> --startLabel <span class="string">&quot;Peak Start&quot;</span> --endLabel <span class="string">&quot;Peak End&quot;</span> --xAxisLabel <span class="string">&quot;&quot;</span> --regionsLabel <span class="string">&quot;Peaks&quot;</span></span><br></pre></td></tr></table></figure>

<h2 id="Peak-annotation–ChIPseeker"><a href="#Peak-annotation–ChIPseeker" class="headerlink" title="Peak annotation–ChIPseeker"></a>Peak annotation–ChIPseeker</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br></pre></td><td class="code"><pre><span class="line">library<span class="punctuation">(</span>ChIPseeker<span class="punctuation">)</span></span><br><span class="line">library<span class="punctuation">(</span>GenomicFeatures<span class="punctuation">)</span></span><br><span class="line">library<span class="punctuation">(</span>tidyverse<span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line">hg38 <span class="operator">&lt;-</span> loadDb<span class="punctuation">(</span><span class="string">&quot;gencode.v44.sqlite&quot;</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line">peak1 <span class="operator">&lt;-</span> readPeakFile<span class="punctuation">(</span><span class="string">&#x27;sample1_seacr_peak.stringent.bed&#x27;</span><span class="punctuation">)</span></span><br><span class="line">peak2 <span class="operator">&lt;-</span> readPeakFile<span class="punctuation">(</span><span class="string">&#x27;sample2_seacr_peak.stringent.bed&#x27;</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line">peaks <span class="operator">&lt;-</span> <span class="built_in">list</span><span class="punctuation">(</span>sample1 <span class="operator">=</span> peak1<span class="punctuation">,</span> sample2 <span class="operator">=</span> peak2<span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line">plotPeakProf2<span class="punctuation">(</span>peaks<span class="punctuation">,</span> upstream <span class="operator">=</span> <span class="number">3000</span><span class="punctuation">,</span> downstream <span class="operator">=</span> <span class="number">3000</span><span class="punctuation">,</span> conf <span class="operator">=</span> <span class="number">0.95</span><span class="punctuation">,</span></span><br><span class="line">              by <span class="operator">=</span> <span class="string">&quot;gene&quot;</span><span class="punctuation">,</span> type <span class="operator">=</span> <span class="string">&quot;start_site&quot;</span><span class="punctuation">,</span> TxDb <span class="operator">=</span> hg38<span class="punctuation">,</span> facet<span class="operator">=</span><span class="string">&quot;row&quot;</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line">plotPeakProf2<span class="punctuation">(</span>peaks<span class="punctuation">,</span> upstream <span class="operator">=</span> rel<span class="punctuation">(</span><span class="number">0.2</span><span class="punctuation">)</span><span class="punctuation">,</span> downstream <span class="operator">=</span> rel<span class="punctuation">(</span><span class="number">0.2</span><span class="punctuation">)</span><span class="punctuation">,</span></span><br><span class="line">              conf <span class="operator">=</span> <span class="number">0.95</span><span class="punctuation">,</span> by <span class="operator">=</span> <span class="string">&quot;gene&quot;</span><span class="punctuation">,</span> type <span class="operator">=</span> <span class="string">&quot;body&quot;</span><span class="punctuation">,</span> nbin <span class="operator">=</span> <span class="number">800</span><span class="punctuation">,</span></span><br><span class="line">              TxDb <span class="operator">=</span> hg38<span class="punctuation">,</span> facet<span class="operator">=</span><span class="string">&quot;row&quot;</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line">peakAnnoList <span class="operator">&lt;-</span> lapply<span class="punctuation">(</span>peaks<span class="punctuation">,</span> annotatePeak<span class="punctuation">,</span> TxDb<span class="operator">=</span>hg38<span class="punctuation">,</span></span><br><span class="line">                       tssRegion<span class="operator">=</span><span class="built_in">c</span><span class="punctuation">(</span><span class="operator">-</span><span class="number">3000</span><span class="punctuation">,</span> <span class="number">3000</span><span class="punctuation">)</span><span class="punctuation">,</span> annoDb<span class="operator">=</span><span class="string">&quot;org.Hs.eg.db&quot;</span><span class="punctuation">)</span></span><br><span class="line">write.table<span class="punctuation">(</span>peakAnnoList<span class="punctuation">[</span><span class="number">1</span><span class="punctuation">]</span><span class="punctuation">,</span> file <span class="operator">=</span> <span class="string">&#x27;sample1_peak.txt&#x27;</span><span class="punctuation">,</span> sep <span class="operator">=</span> <span class="string">&#x27;\t&#x27;</span><span class="punctuation">,</span> <span class="built_in">quote</span> <span class="operator">=</span> <span class="literal">FALSE</span><span class="punctuation">,</span> row.names <span class="operator">=</span> <span class="literal">FALSE</span><span class="punctuation">)</span></span><br><span class="line">write.table<span class="punctuation">(</span>peakAnnoList<span class="punctuation">[</span><span class="number">2</span><span class="punctuation">]</span><span class="punctuation">,</span> file <span class="operator">=</span> <span class="string">&#x27;sample2_peak.txt&#x27;</span><span class="punctuation">,</span> sep <span class="operator">=</span> <span class="string">&#x27;\t&#x27;</span><span class="punctuation">,</span> <span class="built_in">quote</span> <span class="operator">=</span> <span class="literal">FALSE</span><span class="punctuation">,</span> row.names <span class="operator">=</span> <span class="literal">FALSE</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line">plotAnnoBar<span class="punctuation">(</span>peakAnnoList<span class="punctuation">)</span></span><br><span class="line">plotDistToTSS<span class="punctuation">(</span>peakAnnoList<span class="punctuation">)</span>  </span><br></pre></td></tr></table></figure>
<ul>
<li>ChIPseeker Reference: <a target="_blank" rel="noopener" href="https://bioconductor.org/packages/release/bioc/vignettes/ChIPseeker/inst/doc/ChIPseeker.html">https://bioconductor.org/packages/release/bioc/vignettes/ChIPseeker/inst/doc/ChIPseeker.html</a></li>
</ul>
</div></article><div class="post-copyright"><div class="post-copyright__author"><span class="post-copyright-meta">Author: </span><span class="post-copyright-info"><a href="mailto:undefined">Giftbear</a></span></div><div class="post-copyright__type"><span class="post-copyright-meta">Link: </span><span class="post-copyright-info"><a href="https://giftbear.github.io/2024/03/15/CUTTag_Pipeline/">https://giftbear.github.io/2024/03/15/CUTTag_Pipeline/</a></span></div><div class="post-copyright__notice"><span class="post-copyright-meta">Copyright Notice: </span><span class="post-copyright-info">All articles in this blog are licensed under <a target="_blank" rel="noopener" href="https://creativecommons.org/licenses/by-nc-sa/4.0/">CC BY-NC-SA 4.0</a> unless stating additionally.</span></div></div><div class="post-meta__tag-list"><a class="post-meta__tags" href="/tags/Epigenome/">Epigenome</a></div><nav id="pagination"><div class="prev-post pull-left"><a href="/2024/04/09/Perl%E5%AE%89%E8%A3%85%E6%A8%A1%E5%9D%97/"><i class="fa fa-chevron-left">  </i><span>Perl安装模块</span></a></div><div class="next-post pull-right"><a href="/2024/02/27/ChIP-Seq_Pipeline/"><span>ChIP-Seq Pipeline</span><i class="fa fa-chevron-right"></i></a></div></nav></div></div><footer><div class="layout" id="footer"><div class="copyright">&copy;2021 - 2024 By Giftbear</div><div class="framework-info"><span>Driven - </span><a target="_blank" rel="noopener" href="http://hexo.io"><span>Hexo</span></a><span class="footer-separator">|</span><span>Theme - </span><a target="_blank" rel="noopener" href="https://github.com/Molunerfinn/hexo-theme-melody"><span>Melody</span></a></div><div class="busuanzi"><script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script><span id="busuanzi_container_page_pv"><i class="fa fa-file"></i><span id="busuanzi_value_page_pv"></span><span></span></span></div></div></footer><i class="fa fa-arrow-up" id="go-up" aria-hidden="true"></i><script src="https://cdn.jsdelivr.net/npm/animejs@latest/lib/anime.min.js"></script><script src="https://cdn.jsdelivr.net/npm/jquery@latest/dist/jquery.min.js"></script><script src="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@latest/dist/jquery.fancybox.min.js"></script><script src="https://cdn.jsdelivr.net/npm/velocity-animate@latest/velocity.min.js"></script><script src="https://cdn.jsdelivr.net/npm/velocity-ui-pack@latest/velocity.ui.min.js"></script><script src="/js/utils.js?version=1.9.1"></script><script src="/js/fancybox.js?version=1.9.1"></script><script src="/js/sidebar.js?version=1.9.1"></script><script src="/js/copy.js?version=1.9.1"></script><script src="/js/fireworks.js?version=1.9.1"></script><script src="/js/transition.js?version=1.9.1"></script><script src="/js/scroll.js?version=1.9.1"></script><script src="/js/head.js?version=1.9.1"></script><script>if(/Android|webOS|iPhone|iPod|iPad|BlackBerry/i.test(navigator.userAgent)) {
  $('#nav').addClass('is-mobile')
  $('footer').addClass('is-mobile')
  $('#top-container').addClass('is-mobile')
}</script></body></html>